extras/cloudsql-multicluster/tls-for-mci.md

```diff
@@ -0,0 +1,117 @@
+# Enable TLS support on the Multi Cluster Ingress
+---
+
+In this section we will enable **TLS** support _(with
+**HTTP to HTTPS** redirection)_ on the **Multi Cluster Ingress** resource that was
+created in the [Multi Cluster Bank of Anthos with Cloud SQL](README.md) guide.
+
+## Prerequisites
+
+- Bank of Anthos deployed in multiple clusters with access to Cloud SQL as shown
+  in the [previous guide](README.md).
+- This guide assumes that all the environment variables and **kubectx**(s) -
+  _cluster1 & cluster2_ - are still valid
+- This guide assumes that all the commands are executed from the same directory
+  as this README ***`(extras/cloudsql-multicluster/tls-for-mci.md)`***
+
+---
+
+## Part 2
+
+1. **Create a static IP address**. This will be used to generate the _self-signed_
+   certificate.
+```sh
+gcloud compute addresses create boa-multi-cluster-ip --global
+export STATIC_IP=`gcloud compute addresses describe boa-multi-cluster-ip --global --format="value(address)"`
+
+echo $STATIC_IP
+```
+
+2. **Create the TLS certificate**
+```sh
+openssl genrsa -out private.key 2048
+openssl req -new -key private.key -out frontend.csr -subj "/CN=${STATIC_IP}"
+openssl x509 -req -days 365 -in frontend.csr -signkey private.key -out public.crt
+```
+
+3. **Install the TLS certificate as a Kubernetes secret in cluster1**. We use
+   _cluster1_ here because that is the **config cluster** for
+   MutliClusterIngress resources.
+```sh
+kubectx cluster1
+kubectl create secret tls frontend-tls-multi --cert=public.crt --key=private.key
+```
+
+4. **Create the yaml configuration for the `FrontendConfig` and the updated `MultiClusterIngress` resource**.
+   The `FrontendConfig` defines the redirection behaviour for HTTP to HTTPS.
+   We then annotate the `MultiClusterIngress` resource with the `FrontendConfig`
+   and the static IP we created. The `MultiClusterIngress` definition is also
+   updated to use the TLS certificate via the Kubernetes secret we created.
+
+```sh
+envsubst < multicluster-ingress-https.template > multicluster-ingress-https.yaml
```

Comment:
Thought: 
I am wondering if `envsubst` and the user of a separate `.template` file is overkill here since:
1. this is the only place `envsubst` is used, 
1. the only variable being replaced is `${STATIC_IP}`, 
1. some readers might not have `envsubst` already installed

I am wondering if we should just ask users to edit  `${STATIC_IP}` manually.
Just a thought. You make the call. I'm guessing my 3rd point might not even be true.

Reply:
I agree to the point that this might be an overkill. However, I think usually being able to just copy paste the commands is a better user-experience than having to `open the file + look for the correct place + make the change + save and exit`. 

That's the main reason I have opted for `envsubtr`. And if I am not mistaken it is available by default in all Linux based OS's/ I might be wrong but I don't remember installing it in my mac and I am pretty sure Linux flavours all have it by default. I would prefer to stick to this unless you have strong thoughts to do without it.